h/search/query.py

```diff
@@ -15,6 +20,34 @@
 OFFSET_MAX = 9800
 
 
+def wildcard_uri_is_valid(uri):
+    """
+    Return True if uri contains wildcards in appropriate places, return False otherwise.
+
+    Wildcards are not permitted in the domain of the uri.
+    """
+    # If uri is in the form of a unique id, the domain will not be parsed correctly but
+    # it is still valid. Check for this special case and return early.
+    if uri.startswith("urn:x-pdf:"):
+        return True
+    # Let urlparse get what it thinks is the domain from the wildcard_uri and then
+    # strip any remaining "*"s and "?"s. If this matches the start of the original
+    # uri then the uri, the uri is valid.
+    domain = '{uri.scheme}://{uri.netloc}/'.format(
+        uri=urlparse.urlparse(uri)).replace("*", "").replace("?", "")
+    if uri.startswith(domain):
+        return True
+    return False
+
+
+def _pop_param_values(params, key):
```

Comment:
This was added here because I would like to refactor all classes in here to use this (if they need to) in a follow up PR.